src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +230,23 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param types      The types of logs being requested. This should correspond to values in the enum MTRDiagnosticLogType.
+ * @param timeout    The timeout in minutes for getting the logs.  If the timeout expires, completion
+ *                   will be called with whatever logs have been retrieved by that point
+ *                   (which might be none, and might include partial logs).
+ *                   If the timeout is nil, it will never expire and completion will not be called until
+ *                   the logs are fully retrieved.
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to pass in the file paths for the requested logs.
+ */
+- (void)getDiagnosticLogsOfTypes:(NSArray<NSNumber *> *)types
+                         timeout:(NSNumber * _Nullable)timeout
+                           queue:(dispatch_queue_t)queue
+                      completion:(void (^)(NSError * _Nullable error, MTRDiagnosticLogResult * _Nullable logResult))completion MTR_PROVISIONALLY_AVAILABLE;
```

Comment:
The completion parameters should be reversed, result before error.

Reply:
merged the error into the MTRDiagnosticLogResult